### ISSUE TITLE
Unable to build this on centos7

### ISSUE THREAD

**Issue by normalscene** [theStack/bluecube#1]
Hello There,

I am interested to checkout your game but I am not able to build it and I am facing the below shown error. Can you please point me to the right direction so that I could resolve the said issue?

Please let me know if you need additional info to assess the shown problem. I am happy to provide so. Thank you.

```
(20200522-22:02:45) $:bluecube>  sudo rpm -qa | grep SDL2
SDL2_image-2.0.5-1.el7.x86_64
SDL2_image-debuginfo-2.0.5-1.el7.x86_64
SDL2_net-devel-2.0.1-2.el7.x86_64
SDL2_gfx-1.0.3-1.el7.x86_64
SDL2-static-2.0.10-1.el7.x86_64
SDL2_net-debuginfo-2.0.1-2.el7.x86_64
SDL2_mixer-devel-2.0.4-2.el7.x86_64
SDL2_ttf-2.0.15-1.el7.x86_64
SDL2_net-2.0.1-2.el7.x86_64
SDL2_mixer-2.0.4-2.el7.x86_64
SDL2-debuginfo-2.0.10-1.el7.x86_64
SDL2_mixer-debuginfo-2.0.4-2.el7.x86_64
SDL2_image-devel-2.0.5-1.el7.x86_64
SDL2_ttf-devel-2.0.15-1.el7.x86_64
SDL2_gfx-debuginfo-1.0.3-1.el7.x86_64
SDL2_gfx-devel-1.0.3-1.el7.x86_64
SDL2-devel-2.0.10-1.el7.x86_64
SDL2_ttf-debuginfo-2.0.15-1.el7.x86_64
SDL2-2.0.10-1.el7.x86_64


(20200522-22:02:56) $:bluecube>  make
make -C src ../bluecube
make[1]: Entering directory `/home/compuser/mystuff/study/games-2020MAY26/bluecube/src'
gcc -o ../bluecube main.o box.o pieces.o credits.o grafix.o font.o particle.o sound.o mainmenu.o `sdl-config --libs`
/usr/bin/ld: particle.o: undefined reference to symbol 'cos@@GLIBC_2.2.5'
/usr/lib64/libm.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[1]: *** [../bluecube] Error 1
make[1]: Leaving directory `/home/compuser/mystuff/study/games-2020MAY26/bluecube/src'
make: *** [bluecube] Error 2


(20200522-22:02:58) $:bluecube>  uname -a
Linux lenovoe470.localdomain 3.10.0-1127.el7.x86_64 #1 SMP Tue Mar 31 23:36:51 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux


(20200522-22:03:04) $:bluecube>  cat /etc/redhat-release 
CentOS Linux release 7.8.2003 (Core)
(20200522-22:03:10) $:bluecube>  


(20200522-22:03:10) $:bluecube>  g++ -v
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/opt/mystuff/compilers/gcc-9.2.0/bin/../libexec/gcc/x86_64-linux-gnu/9.2.0/lto-wrapper
Target: x86_64-linux-gnu
Configured with: configure -v --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu --prefix=/opt/gcc-9.2.0 --enable-languages=c,c++,fortran --disable-multilib
Thread model: posix
gcc version 9.2.0 (GCC) 
(20200522-22:03:16) $:bluecube>  
```

**Comment by theStack** [comment]
Hi normalscene,
thanks for your report! I just tried it out on myself on my system (Lubuntu) and get the same error. Seems that linking to the standard math library is missing (I have absolutely no idea why back then, over 10 years ago when I used this the last time, this was not needed :smile:). On the following line in `src/Makefile`

https://github.com/theStack/bluecube/blob/63071a4b97032c40bd4885347cee0fc3b1337e96/src/Makefile#L6

a `-lm` has to be added, and it should work.

That's the quick fix, but will push a commit later anyway.

Cheers,
theStack
